### PR TITLE
Make CBMC proofs follow release-1.5 directory layout

### PR DESCRIPTION
**Pull request by karkhaz**
This PR changes the paths in all of the CBMC proof configuration files so that the proofs can run on the release-1.5 branch. The release-1.5 branch significantly changed the directory structure, meaning that the CBMC proofs could not link against FreeRTOS object files. This PR is in anticipation of merging the release-1.5 branch into master, so that the proofs will continue to run on the master branch.

### CODE REVIEW

File: tools/cbmc/patches/0001-Remove-static-storage-class-from-entry-points.patch [hunk 1]
```diff
@@ -7,15 +7,15 @@ Many of the entry points we wish to test are marked as being static.
 This commit removes the static keyword from all entry points that we
 test.
 ---
- lib/FreeRTOS-Plus-TCP/source/FreeRTOS_DHCP.c    |  6 +++---
- lib/FreeRTOS-Plus-TCP/source/FreeRTOS_DNS.c     | 14 +++++++-------
- lib/FreeRTOS-Plus-TCP/source/FreeRTOS_TCP_WIN.c |  2 +-
+ modules/libraries/standard/freertos_plus_tcp/source/FreeRTOS_DHCP.c    |  6 +++---
```
Comment (qiutongs) [inline tools/cbmc/patches/0001-Remove-static-storage-class-from-entry-points.patch]: we don't have "modules" anymore. The top directories are libraries and freertos_kernel

Comment (karkhaz) [reply tools/cbmc/patches/0001-Remove-static-storage-class-from-entry-points.patch]: Thanks, I'm on it (good thing the merge conflicted)

**Comment by karkhaz** [comment]
This isn't going to be finished tonight. I'll hopefully finish tomorrow morning and have it ready to merge by morning Seattle time.

File: tools/cbmc/proofs/make_common_makefile.py [hunk 2]
```diff
@@ -195,16 +195,16 @@ def write_cbmcbatchyaml_target(opsys, _makefile):
...
-'=$(shell echo $(1) | sed 's/ ,/ /g' | sed 's/ /;/g')='
+	'=$(shell echo $(1) | sed 's/ ,/ /g' | sed 's/ /;/g')='
 endef
 
 cbmc-batch.yaml:
 	@echo "Building $@"
 	@$(RM) $@
 	@echo "jobos: ubuntu16" >> $@
-	@echo "cbmcflags: $(call encode_options,$(CBMCFLAGS) --unwinding-assertions)" >> $@
+	@echo "cbmcflags: $(call encode_options,$(CBMCFLAGS))" >> $@
```
Comment (mmuesly) [inline tools/cbmc/proofs/make_common_makefile.py]: I think unwinding-assertions should always be checked. Is there a specific reason to remove this flag? At least I would expect it to appear than in the CBMCFLAGS variable.

File: tools/cbmc/proofs/make_common_makefile.py [hunk 3]
```diff
@@ -195,16 +195,16 @@ def write_cbmcbatchyaml_target(opsys, _makefile):
 # Build configuration file to run cbmc under cbmc-batch in CI
 
 define encode_options
-'=$(shell echo $(1) | sed 's/ ,/ /g' | sed 's/ /;/g')='
+	'=$(shell echo $(1) | sed 's/ ,/ /g' | sed 's/ /;/g')='
```
Comment (mmuesly) [inline tools/cbmc/proofs/make_common_makefile.py]: Why is this required for back porting? Other PR?

File: tools/cbmc/proofs/TCP_parsing/ParseDNSReply/Makefile.json [hunk 4]
```diff
@@ -43,14 +43,14 @@
   "CBMCFLAGS":
   [
     "--unwind 1",
-    "--unwindset {PARSELOOP0}:{PARSELOOP0_UNWIND},{PARSELOOP1}:{PARSELOOP1_UNWIND},prvProcessDNSCache.0:5"
+    "--unwindset {PARSELOOP0}:{PARSELOOP0_UNWIND},{PARSELOOP1}:{PARSELOOP1_UNWIND}"
```
Comment (mmuesly) [inline tools/cbmc/proofs/TCP_parsing/ParseDNSReply/Makefile.json]: Is this related to back porting?

File: tools/cbmc/proofs/ARP/ARPAgeCache/cbmc-batch.yaml [hunk 5]
```diff
@@ -1,4 +0,0 @@
-jobos: ubuntu16
```
Comment (mmuesly) [inline tools/cbmc/proofs/ARP/ARPAgeCache/cbmc-batch.yaml]: I know that we have to remove the cbmc-batch.yaml files from the repository, but I think this should be done in another PR. I don't see how they interfere with the pure back porting.
There are no paths in the cbmc-batch.yaml files.

File: tools/cbmc/proofs/TCP_parsing/ProcessDHCPReplies/Makefile.json [hunk 6]
```diff
@@ -18,7 +18,7 @@
 
 ################################################################
 # Buffer payload
-  "BUFFER_PAYLOAD": "__eval 1 if {BUFFER_SIZE} <= {BUFFER_HEADER} else {BUFFER_SIZE} - {BUFFER_HEADER} + 2",
+  "BUFFER_PAYLOAD": "__eval 1 if {BUFFER_SIZE} <= {BUFFER_HEADER} else {BUFFER_SIZE} - {BUFFER_HEADER}",
```
[the remark this answers is not in the document]
Comment (karkhaz) [reply tools/cbmc/proofs/TCP_parsing/ProcessDHCPReplies/Makefile.json]: Reverted to master.

File: tools/cbmc/proofs/make_common_makefile.py [hunk 7]
```diff
@@ -195,16 +195,16 @@ def write_cbmcbatchyaml_target(opsys, _makefile):
...
 
 cbmc-batch.yaml:
 	@echo "Building $@"
 	@$(RM) $@
 	@echo "jobos: ubuntu16" >> $@
-	@echo "cbmcflags: $(call encode_options,$(CBMCFLAGS) --unwinding-assertions)" >> $@
+	@echo "cbmcflags: $(call encode_options,$(CBMCFLAGS))" >> $@
 	@echo "goto: $(ENTRY).goto" >> $@
-	@echo "expected: $(H_EXPECTED)" >> $@
+	@echo "expected: SUCCESSFUL" >> $@
```
[the remark this answers is not in the document]
Comment (karkhaz) [reply tools/cbmc/proofs/make_common_makefile.py]: Reverted to master.

File: tools/cbmc/proofs/make_common_makefile.py (hunk 2 above)
Comment (karkhaz) [reply tools/cbmc/proofs/make_common_makefile.py]: Reverted to master.

File: tools/cbmc/proofs/TCP_parsing/ParseDNSReply/Makefile.json (hunk 4 above)
Comment (karkhaz) [reply tools/cbmc/proofs/TCP_parsing/ParseDNSReply/Makefile.json]: Reverted to master.

File: tools/cbmc/proofs/ARP/ARPAgeCache/cbmc-batch.yaml (hunk 5 above)
Comment (karkhaz) [reply tools/cbmc/proofs/ARP/ARPAgeCache/cbmc-batch.yaml]: Thank you, I'm keeping all `cbmc-batch.yaml` files for this PR now.

File: tools/cbmc/proofs/make_common_makefile.py (hunk 3 above)
Comment (karkhaz) [reply tools/cbmc/proofs/make_common_makefile.py]: Reverted to master.

**Comment by karkhaz** [comment]
@mmuesly thank you for your comments, I've reduced the scope of this PR as you suggested. This PR now includes only path fixups for the existing proofs, as well as the top-level build files. I'll post unrelated changes as separate PRs.

**Comment by karkhaz** [comment]
FWIW, I used the following script to validate that all the updated paths referenced from the `Makefile.json`s actually exist in the source tree. The CI should also tell us whether the renaming was successful.
```python
#!/usr/bin/env python3

import os
import os.path
import json
import re


def check(data, fyle_name):
    if isinstance(data, list):
        for item in data:
            check(item, fyle_name)
    elif isinstance(data, dict):
        for item in data.values():
            check(item, fyle_name)
    elif isinstance(data, str):
        if not data.startswith("$(FREERTOS)"):
            return
        data = data[len("$(FREERTOS)") + 1:]
        if data.endswith(".goto"):
            data = "%s.c" % data[:-len(".goto")]
        if not os.path.exists("../../../%s" % data):
            print(fyle_name)
            print("    %s" % data)


for root, _, fyles in os.walk("."):
    for fyle in fyles:
        if os.path.splitext(fyle)[1] != ".json":
            continue
        with open(os.path.join(root, fyle)) as f:
            tmp = f.read().splitlines()
        buf = [l for l in tmp if l and l.strip()[0] != "#"]
        doc = "\n".join(buf)
        data = json.loads(doc)

        check(data, os.path.join(root, fyle))
```

File: tools/cbmc/proofs/MakefileCommon.json [hunk 8]
```diff
@@ -13,27 +13,23 @@
...
+	"$(FREERTOS)/vendors/pc/boards/windows/aws_demos/application_code",
+	"$(FREERTOS)/demos/include",
+	"$(FREERTOS)/freertos_kernel/include",
+	"$(FREERTOS)/freertos_kernel/portable/MSVC-MingW",
+	"$(FREERTOS)/libraries/freertos_plus/standard/freertos_plus_tcp",
+	"$(FREERTOS)/libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/BufferManagement",
+	"$(FREERTOS)/libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/Compiler/MSVC",
+	"$(FREERTOS)/libraries/freertos_plus/aws/ota/include",
+	"$(FREERTOS)/libraries/3rdparty/mbedtls/include/mbedtls",
+	"$(FREERTOS)/libraries/3rdparty/tracealyzer_recorder",
```
Comment (mmuesly) [inline tools/cbmc/proofs/MakefileCommon.json]: I think the new path should be `$(FREERTOS)/libraries/3rdparty/tracealyzer_recorder/include`

File: tools/cbmc/patches/0003-Replace-MiniListItem_t-with-ListItem_t.patch [hunk 9]
```diff
@@ -11,13 +11,13 @@ element of lists a ListItem_t instead.
 
 This patch should be removed soon: It is for a CBMC issue being fixed now.
 ---
- lib/include/private/list.h | 2 +-
+ freertos_kernel/list.h | 2 +-
```
Comment (mmuesly) [inline tools/cbmc/patches/0003-Replace-MiniListItem_t-with-ListItem_t.patch]: `freertos_kernel/include` instead?

**Comment by qiutongs** [comment]
Today, we completed our new release. And we have merged release-1.5 to master. If it is OK, can you make this PR base off master branch?
